### PR TITLE
Backend logic for request returns - issue #52

### LINKED ISSUE #52: Develop returning book logic
**Description**
Develop backend logic for returning a book
1. User should be able to request to return a book
2. User should be able to cancel a request to return a book
3. User should be able to return a book

### PR DESCRIPTION
**Pull request by akchen1**
User should be able to request to return a book
User should be able to cancel a request to return a book
User should be able to return a book

### CODE REVIEW

File: app/src/main/java/com/team41/boromi/controllers/BookReturnController.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+import java.util.concurrent.Executor;
+import javax.inject.Inject;
+
+public class BookReturnController {
+
+  private final static String TAG = "ReturnController";
+  final Executor executor;
+  BookReturnDB returnDB;
+  BookDB bookDB;
+  FirebaseAuth auth;
```
Comment (mingyaang) [inline app/src/main/java/com/team41/boromi/controllers/BookReturnController.java]: why do you need firebase auth

Comment (mingyaang) [reply app/src/main/java/com/team41/boromi/controllers/BookReturnController.java]: Also annotate this controller with @ singleton so it is not recreated in each activity/fragment

**Review by mingyaang — APPROVED** [review]
This is good code ! Please make the slight changes :) 👍

File: app/src/main/java/com/team41/boromi/controllers/BookReturnController.java (hunk 1 above)
Comment (akchen1) [reply app/src/main/java/com/team41/boromi/controllers/BookReturnController.java]: fixed!

**Pull request merged**